streaming/aggregate/sum.go

```diff
@@ -9,20 +9,37 @@ import (
 	"github.com/cube2222/octosql/streaming/storage"
 )
 
+var currentSumPrefix = []byte("$current_sum$")
+
 type Sum struct {
 }
 
-var currentSumPrefix = []byte("$current_sum$")
+func NewSumAggregate() *Sum {
+	return &Sum{}
+}
 
 func (agg *Sum) AddValue(ctx context.Context, tx storage.StateTransaction, value octosql.Value) error {
 	currentSumStorage := storage.NewValueState(tx.WithPrefix(currentSumPrefix))
 
+	valueType := value.GetType()
+
 	currentSum, err := agg.GetValue(ctx, tx)
 	if err != nil {
 		return errors.Wrap(err, "couldn't get current sum")
 	}
 
-	switch valueType := value.GetType(); valueType {
+	if isNeutralElement(currentSum) {
```

Comment:
This is kind of confusing... Does this mean we can sum floats, reach zero, and then start summing ints? That's what it looks like, since if we reach 0.0 for example, then we change currentSum to 0 (getAppropriateNeutralElement) and we can freely sum ints. I don't know if this was the desired effect

Reply:
It's the thing I actually noticed (yet forgot to point out) and had pretty hard time to workaround. But it's a little bit different - You get appropriate neutral element only if GetValue returns MakeInt(0), which *should* happen only when there is no value in storage (ErrNotFound -> during first AddValue call). Which is a good idea, because types will match. BUT. Lets say the order looks like that:
1) Add(3)
2) Retract(3) (Here, the storage is NOT empty, but GetValue actually returns 0 which shouldn't be preceived as neutral element in this case, but as an actual sum of some elements)
3) Add(123.123) (This will work, because AddValue believes, that 0 returned by GetValue is a neutral element, and this function call is the first one -> which is actually not true)

This works only when Integers are summed in the first place, and I understand (and I did back when writing this) that it has to be fixed (e.x. by storing currentSum type in storage instead of asking "is current sum equal to neutral element". What do you think about it? The idea is quite the same as with "Min/Max type check" comment above
@cube2222 <--- another thing to discuss

Reply:
You can also have negative integers which summed up return zero.